src/main/java/cpsc/dlsproject/visitors/ParseVisitor.java

```diff
@@ -0,0 +1,289 @@
+        }
+        varDeclaration.name = name.substring(0, name.length() - 1);
+
+        String type = tokenizer.getNext();
+
+        switch (type) {
+            case "Number":
+                varDeclaration.type = Type.NUMBER;
+                varDeclaration.expression = new NumberValue(Double.parseDouble(tokenizer.getNext()));
+                break;
+
+            case "Boolean":
+                varDeclaration.type = Type.BOOLEAN;
+                String token = tokenizer.getNext();
+                if (!token.equals("true") && !token.equals("false")) {
+                    throw new Exception("Boolean type is not either true or false.");
+                }
+                varDeclaration.expression = new BooleanValue(token.equals("true"));
+                break;
+
+            default:
+                throw new Exception("Invalid type in Var declaration.");
+
+        }
+
+//        List<String> expressionList = tokenizer.getNextExpression();
+//
+//
+//        String expression = InfixToPrefixConverter.infixToPrefix(expressionList);
+//        currentExpression = expression;
+//
+//        System.out.println(expression);
+//
+//        int i = 0;
+//        while(i < expression.length()) {
+//            if (utils.checkIfTokenIsNumber(expression.charAt(i) + "")) {
+//                String numberStr = "";
+//                while (i < expression.length() && (utils.checkIfTokenIsNumber(expression.charAt(i) + "") || expression.charAt(i) == '.')) {
+//                    numberStr += expression.charAt(i);
+//                    i++;
+//                }
+//                currentExpression = currentExpression;
+//            }
+//            i++;
+//        }
+
+        return varDeclaration;
+    }
+
+    @Override
+    BaseAST visit(BinaryOperation binOp) throws Exception {
+        String token = tokenizer.getNext();
+        if (token.equals("+")) {
+            binOp.operator = BinaryOperator.PLUS;
+        } else if (token.equals("-")) {
+            binOp.operator = BinaryOperator.MINUS;
+        } else if (token.equals("*")) {
+            binOp.operator = BinaryOperator.MULTIPLY;
+        } else if (token.equals("\\")) {
+            binOp.operator = BinaryOperator.DIVISION;
+        } else if (token.equals("%")) {
+        } else if (token.equals("<")) {
+            binOp.operator = BinaryOperator.LESSER;
+        } else if (token.equals("<=")) {
+            binOp.operator = BinaryOperator.LEQUAL;
+        } else if (token.equals("==")) {
+            binOp.operator = BinaryOperator.EQUAL;
+        } else if (token.equals("!=")) {
+            binOp.operator = BinaryOperator.NOTEQUAL;
+        } else if (token.equals(">=")) {
+            binOp.operator = BinaryOperator.GEQUAL;
+        } else if (token.equals(">")) {
+            binOp.operator = BinaryOperator.GREATER;
+        } else {
+            throw new Exception(token + " cannot be parsed as expression.");
+        }
+
+        binOp.lhs = parseExpression();
```

Comment:
TODO:: Add this change to EBNF as well as the Language Design Specification Changelist